tests/components/test_power_output.py

```diff
@@ -88,3 +92,35 @@ def test_power_output_current() -> None:
     power_output = PowerOutput(0, MockPowerOutputBoard(), MockPowerOutputDriver())
     assert type(power_output.current) is float
     assert power_output.current == 8.1
+
+
+def test_power_output_group_instantiation() -> None:
+    """Test that we can instantiate a PowerOutput group."""
+    board = MockPowerOutputBoard()
+    outputs = {i: PowerOutput(i, board, MockPowerOutputDriver()) for i in range(0, 5)}
```

Comment:
Considering that this PR also changes `MockPowerOutputDriver` to make it behave independently when different `identifier`s are given, I think that this line should use the same instance of `MockPowerOutputDriver` for all outputs rather than creating new ones each time.

```suggestion
    backend = MockPowerOutputDriver(5)
    outputs = {i: PowerOutput(i, board, backend) for i in range(0, 5)}
```
and similarly for the other two test cases.